utils/codegen/codegen-py/asn1CodeGenerationUtils.py

```diff
@@ -449,12 +445,18 @@ def asn1TypeToJinjaContext(asn1_type_name: str, asn1_type_info: Dict, asn1_types
     }
 
     # extra information / asn1 fields that are not processed as comments
-    for k, v in asn1_type_info.items():
-        if k not in ("type", "element", "members", "name", "named-bits", "named-numbers", "optional", "restricted-to", "size", "values", "default"):
-            context["comments"].append(f"{k}: {v}")
+    if not isinstance(asn1_type_info, list):
+        for k, v in asn1_type_info.items():
+            if k not in ("type", "element", "members", "name", "named-bits", "named-numbers", "optional", "restricted-to", "size", "values", "default"):
+                context["comments"].append(f"{k}: {v}")
+
+    # components-of
+    if "components-of" in asn1_type_info:
```

Comment:
Do we need to handle this as the first condition? If not, would move it down.

Reply:
We need to handle this at the beginning.